### DISCUSSION TITLE
Need info about bind function. (multi arguments)

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2188] [Q&A] [answered]
Hi everyone,

I would to use few argument with a bind function.

It seem that the lambda why generate a bug in my test.

```python
for each in document.select('#TestDIV textarea'):
	each.bind("input", lambda ev: TheFunction(ev, what=each.name , aVariable=1000))
```
The error here is that all the what variable have the same output in each loop :/

So instead of a lambda is there a more classical why ?

I've tried few other but it always fail because I need to pass the `ev` but I don't know how to get it..

**Comment by PierreQuentel** [comment, marked as answer]
Hi !

This kind of issue (which is not specific to Brython) is solved by passing the loop variable to the lambda function:

```python
from browser import document

def f(ev, what):
    print(what)
    
for each in document.select('div'):
    each.bind("click", lambda ev, each=each: f(ev, what=each.name))
```

**Reply by BobSquarePants** [reply]
Thank you @PierreQuentel ,

As said, is there another way (more classical) than a `lambda ` ?

Thanks.

**Comment by PierreQuentel** [comment]
In this case, `lambda` _is_ the "classical way". I borrowed the name `bind` from Tkinter, in which `lambda` is used to pass arguments to callback functions. See for instance this [stackoverflow question](https://stackoverflow.com/questions/33276717/how-can-i-pass-an-argument-to-a-tkinter-bind-callback).
